src/jsonsync.py

```
@@ -0,0 +1,101 @@
+        self.filename = filename
+    
+    def syncbase(self, key):
+        if self.base is None:
+            if type(key) is int:
+                self.base = list()
+            else:
+                self.base = dict()
+
+    def __getitem__(self, key):
+        self.syncbase(key)
+        assert self.base is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)